### PR TITLE
Added tests for org.locationtech.geowave.core.index.ByteArrayUtils

### PR DESCRIPTION
**Pull request by freddytuxworth**
Hi,
I've analysed this codebase and noticed some gaps in the coverage of `ByteArrayUtils.java`.

I've written a few tests with the help of [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/datasheet), so that this class is fully tested.

Hopefully these tests should help you detect regressions caused by future code changes. Please let me know if you'd be interested in having more tests written for other classes within this project, I would be happy to help.

### CODE REVIEW

File: core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java [hunk 1]
```diff
@@ -38,9 +40,121 @@ public void testVariableLengthEncodeDecode() {
     testVariableLengthValue(Long.MIN_VALUE);
   }
 
+  @Test
+  public void testCombineArrays() {
+    Assert.assertArrayEquals(new byte[] {1,2,3,4}, ByteArrayUtils.combineArrays(new byte[] {1,2,3,4}, null));
```
Comment (peterschrammel) [inline core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java]: Suggest to break lines
```
Assert.assertArrayEquals(
  new byte[] {1,2,3,4},
  ByteArrayUtils.combineArrays(new byte[] {1,2,3,4}, null));
```

File: core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java [hunk 2]
```diff
@@ -38,9 +40,121 @@ public void testVariableLengthEncodeDecode() {
...
+  @Test
+  public void testByteArrayFromString() {
+    Assert.assertArrayEquals(new byte[]{(byte) -75, (byte) -21, (byte) 45}, ByteArrayUtils.byteArrayFromString("test"));
+    Assert.assertArrayEquals(new byte[]{}, ByteArrayUtils.byteArrayFromString(""));
+  }
+
+  @Test
+  public void testIncrement() {
+    byte[] testByte = new byte[]{(byte) -1};
+    Assert.assertEquals(false, ByteArrayUtils.increment(testByte));
```
Comment (peterschrammel) [inline core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java:88]: add newlines to separate the first line and the two test cases

File: core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java [hunk 3]
```diff
@@ -38,9 +40,121 @@ public void testVariableLengthEncodeDecode() {
...
+    Assert.assertEquals(0, ByteArrayUtils.compare(new byte[]{(byte) 0}, new byte[]{(byte) 0}));
+    Assert.assertEquals(1, ByteArrayUtils.compare(null, new byte[]{(byte) 0}));
+    Assert.assertEquals(0, ByteArrayUtils.compare(null, null));
+    Assert.assertEquals(0, ByteArrayUtils.compare(null, null));
+  }
+
+
+  @Test
+  public void testByteArrayFromString() {
+    Assert.assertArrayEquals(new byte[]{(byte) -75, (byte) -21, (byte) 45}, ByteArrayUtils.byteArrayFromString("test"));
```
Comment (peterschrammel) [inline core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java]: Are these casts necessary?

File: core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java [hunk 4]
```diff
@@ -38,9 +40,152 @@ public void testVariableLengthEncodeDecode() {
...
+    Assert.assertArrayEquals(new byte[]{}, ByteArrayUtils.byteArrayFromString(""));
+  }
+
+  @Test
+  public void testIncrement() {
+    byte[] testByte = new byte[]{(byte) -1};
+
+    Assert.assertEquals(false, ByteArrayUtils.increment(testByte));
+    Assert.assertArrayEquals(new byte[] {(byte) 0}, testByte);
+    Assert.assertEquals(true, ByteArrayUtils.increment(testByte));
```
Comment (peterschrammel) [inline core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java:91]: I'd also separate the last two lines

File: core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java [hunk 5]
```diff
@@ -38,9 +40,152 @@ public void testVariableLengthEncodeDecode() {
...
     byte[] encoded = ByteArrayUtils.variableLengthEncode(value);
     long result = ByteArrayUtils.variableLengthDecode(encoded);
     Assert.assertEquals(value, result);
   }
+
+  @Test
+  public void testCompare() {
+    Assert.assertEquals(-1, ByteArrayUtils.compare(new byte[]{}, null));
+    Assert.assertEquals(0, ByteArrayUtils.compare(new byte[]{}, new byte[]{}));
+    Assert.assertEquals(1, ByteArrayUtils.compare(new byte[]{(byte) 0}, new byte[]{}));
```
Comment (peterschrammel) [inline core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java]: There are many more casts that could be dropped.

File: core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java [hunk 6]
```diff
@@ -38,9 +40,152 @@ public void testVariableLengthEncodeDecode() {
     testVariableLengthValue(Long.MIN_VALUE);
   }
 
+  @Test
+  public void testCombineArrays() {
+    Assert.assertArrayEquals(
+            new byte[] {1,2,3,4},
```
Comment (peterschrammel) [inline core/index/src/test/java/org/locationtech/geowave/core/index/ByteArrayUtilsTest.java]: Run the code through the formatter.

**Review by paulkennethkent — APPROVED** [review]

**Review by sudo-jaa — APPROVED** [review]

**Pull request closed without merging**